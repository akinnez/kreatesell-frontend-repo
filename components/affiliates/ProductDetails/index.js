import {useRouter} from 'next/router';

import {Button} from 'antd';

import Performance from 'components/affiliates/Performance';
import styles from './index.module.scss';
import {RenderIf} from 'utils';

const ProductDetails = ({
	kreatorName,
	productType,
	currency,
	price,
	sold,
	visits,
	commission,
	productId = '',
}) => {
	const router = useRouter();
	return (
		<>
			<ol>
				<li className={styles.product__detail}>
					<strong>Kreator</strong>
					<span>{kreatorName}</span>
				</li>
				<li className={styles.product__detail}>
					<strong>Product Type</strong>
					<span>{productType}</span>
				</li>
				<li className={styles.product__detail}>
					<strong>Price</strong>
					<span>
						{currency} {price}
					</span>
				</li>
				<li
					className={`${styles.product__detail} ${styles.product__performance}`}
				>
					<strong>Performance</strong>
					<span>
						<Performance sold={sold} visits={visits} />
					</span>
				</li>
				<li className={styles.product__detail}>
					<strong>Commission (%)</strong>
					<span>{commission}%</span>
				</li>
			</ol>
			<RenderIf condition={router.pathname.includes('market-place')}>
				<div className="my-5 text-center">
					<Button
						className={`${styles.viewProductBtn} px-5`}
						type="primary"
						htmlType="button"
						onClick={() =>
							router.push(
								`/account/affiliate/market-place/${productId}?activeTab=overview`
							)
						}
					>
						View Product
					</Button>
				</div>
			</RenderIf>
		</>
	);
};

export default ProductDetails;
