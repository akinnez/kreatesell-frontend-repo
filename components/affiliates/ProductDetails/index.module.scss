@use "public/css/colors";
@use "public/css/mixins";
@use "public/css/screens";

.product__detail {
  padding: 6px 10px;
  @include mixins.flex(row, flex-start, center);
  flex-wrap: wrap;
  gap: 2px 70px;

  &:nth-child(odd) {
    border-radius: 8px;
    background-color: colors.$white-200;
  }

  &:not(:last-of-type) {
    margin-bottom: 10px;
  }

  strong {
    width: 115px;
    font-size: 14px;
    color: colors.$black-100;
  }

  &:not([class*="product__performance"]) span {
    font-size: 14px;
    color: colors.$base-gray;
  }
}

.product__performance span {
  font-size: 14px;
  display: inline-flex;
  column-gap: 15px;
}

@include screens.screen("medium") {
  .product__detail {
    padding-top: 4px;
    padding-bottom: 4px;
    column-gap: 10px;
  }
}
