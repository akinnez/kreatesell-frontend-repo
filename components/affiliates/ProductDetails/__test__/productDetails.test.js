import ProductDetails from "..";
import renderer from "react-test-renderer";

const mockProps = {
  kreatorName: "mock kreatorName",
  productType: "digital product",
  currency: "NGN",
  price: 2500,
  sold: 250,
  visits: 20,
  commission: 10,
};

it("renders a snapshot of <ProductDetails /> with mock props", () => {
  const details = renderer.create(<ProductDetails {...mockProps} />).toJSON();
  expect(details).toMatchSnapshot();
});
