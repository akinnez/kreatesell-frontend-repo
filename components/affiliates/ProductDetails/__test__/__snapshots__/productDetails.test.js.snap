// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <ProductDetails /> with mock props 1`] = `
<ol>
  <li
    className="product__detail"
  >
    <strong>
      Kreator
    </strong>
    <span>
      mock kreatorName
    </span>
  </li>
  <li
    className="product__detail"
  >
    <strong>
      Product Type
    </strong>
    <span>
      digital product
    </span>
  </li>
  <li
    className="product__detail"
  >
    <strong>
      Price
    </strong>
    <span>
      NGN
       
      2500
    </span>
  </li>
  <li
    className="product__detail product__performance"
  >
    <strong>
      Performance
    </strong>
    <span>
      <div>
        <span
          className="sold"
        >
          Sold:
        </span>
         
        250
      </div>
      <div>
        <span
          className="visits"
        >
          Visit:
        </span>
         
        20
      </div>
    </span>
  </li>
  <li
    className="product__detail"
  >
    <strong>
      Commission (%)
    </strong>
    <span>
      10
      %
    </span>
  </li>
</ol>
`;
